bugfix_19795

xus opened the pull request:
[http://www.ilias.de/mantis/view.php?id=19795](http://www.ilias.de/mantis/view.php?id=19795)

xus committed:
bugfix_19795

xus committed:
dev notes, bug 19795

xus committed:
Migration files after patch

xus committed:
migration patch for bugfix_19795

xus committed:
delete wrong migration file

alex40724 commented on setup/sql/dbupdate_custom.php:
```diff
@@ -0,0 +1,136 @@
+		$res_crit = $db->query("SELECT id FROM exc_crit ".
+			" WHERE parent = ".$row['peer_crit_cat'].
+			" AND type = 'file'"
+		);
```
This is missing proper quoting

alex40724 commented on setup/sql/dbupdate_custom.php:
```diff
@@ -0,0 +1,136 @@
+	$ilDB->createTable("exc_ass_file_order", $fields);
+	$ilDB->addPrimaryKey("exc_ass_file_order", array("id"));
+	$ilDB->createSequence("exc_ass_file_order", 1);
+}
```
This belongs to the new feature development. I guess I can safely ignore this when merging.

xus replied:
Yes, it belongs to the new feature, you can ignore it.

xus committed:
quoting

xus committed:
global $ilDB instead of $DIC

I had this code in another PHP for testing and I used $DIC. Now I
changed $DIC to global $ilDB to still working like the other updates.

alex40724 commented on setup/sql/dbupdate_custom.php:
```diff
@@ -0,0 +1,136 @@
+		{
+			$original_path = $storage->getPeerReviewUploadPath($row['peer_id'], $row['giver_id'], $row_crit['id']);
+
+			$path_peaces = explode('/', rtrim($a_original_path, '/'));
```
This does not seem to be defined.

alex40724 commented on setup/sql/dbupdate_custom.php:
```diff
@@ -0,0 +1,136 @@
+	{
+		$original_path = $storage->getPeerReviewUploadPath($row['peer_id'], $row['giver_id'], $row['peer_crit_cat']);
+
+		$path_peaces = explode('/', rtrim($a_original_path, '/'));
```
again, see above

alex40724 commented:
Thanks xus. For others: The merge looks very different from these commits (no custom update merged, e.g.).
